client/app/bundles/course/assessment/submission/actions/index.js

```diff
@@ -241,6 +242,72 @@ export function resetAnswer(submissionId, answerId, questionId) {
   };
 }
 
+export function deleteFile(submissionId, answerId, fileId, answerFields) {
+  const answers = Object.values(answerFields);
+  const payload = { submission: { answers } };
+
+  return (dispatch) => {
+    dispatch({ type: actionTypes.DELETE_FILE_REQUEST });
+
+    return CourseAPI.assessment.submissions.update(submissionId, payload)
+      .then(response => response.data)
+      .then((data) => {
+        if (data.redirect_url && data.format === 'html') {
+          window.location = data.redirect_url;
+        }
+        dispatch({ type: actionTypes.DELETE_FILE_SUCCESS, payload: data });
+        dispatch(reset(formNames.SUBMISSION));
```

Comment:
All answers for a submission share a redux-form form. We should be careful when calling redux-form's `reset` as it resets the entire form. If I am not wrong, with this `reset` deleting a file in one question will erase any unsaved changes in other questions.

Instead, you can reset the form values of a single answer inside `client/app/bundles/course/assessment/submission/reducers/form.js`. This would be similar to the cases for `AUTOGRADE_SUCCESS` and `RESET_SUCCESS`